sympy/solvers/tests/test_solvers.py

```diff
@@ -1460,12 +1459,12 @@ def test_issue_5901():
     assert solve([f(x) - 3*f(x).diff(x)], f(x)) == \
         {f(x): 3*D}
     assert solve([f(x) - 3*f(x).diff(x), f(x)**2 - y + 4], f(x), y) == \
-        [{f(x): 3*D, y: 9*D**2 + 4}]
```

Comment:
This should be a dict because even though it is non-linear in `f(x)`, the solution is obtained by linear substitution (solving first for linear `f(x)` and substituting into second to solve for linear `y`).

Reply:
But that is a byproduct piece of information and we are trying to move away from multiple forms of output. So I think this is ok. The symbols were ordered and a system was passed, so a tuple of ordered solutions is returned.